Sources/Networking/HTTPClient.swift

```diff
@@ -187,43 +201,51 @@ private extension HTTPClient {
             return .failure(ErrorUtils.unexpectedBackendResponseError())
         }
 
-        func extractBody(_ response: HTTPURLResponse, _ statusCode: HTTPStatusCode) throws -> [String: Any] {
-            if let data = data, statusCode != .notModified {
-                let result: [String: Any]?
-
-                do {
-                    result = try JSONSerialization.jsonObject(with: data) as? [String: Any]
-                } catch let jsonError {
-                    Logger.error(Strings.network.parsing_json_error(error: jsonError))
-
-                    let dataAsString = String(data: data, encoding: .utf8) ?? ""
-                    Logger.error(Strings.network.json_data_received(dataString: dataAsString))
-
-                    throw jsonError
-                }
-
-                guard let result = result else {
-                    throw ErrorUtils.unexpectedBackendResponseError()
-                }
-
-                return result
+        /// - Returns `nil` if status code is 304, since the response will be empty
+        /// and fetched from the eTag.
+        func dataIfAvailable(_ statusCode: HTTPStatusCode) throws -> Data? {
+            if statusCode == .notModified {
+                return nil
             } else {
-                // No data if the body was empty, or if status is `.notModified`
-                // since the body will be fetched from the eTag.
-                return [:]
+                return data
             }
+
+//            if let data = data, statusCode != .notModified {
```

Comment:
This is one of the last TODOs, when the response fails to deserialize I need to improve how the error is created. Now the `CodableError` is what's forwarded which isn't ideal.

Reply:
Working on it in #1437, that's the last piece to finish.